x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -541,7 +538,8 @@ public Expression visitFunctionExpression(EsqlBaseParser.FunctionExpressionConte
     @Override
     public Expression visitInlineCast(EsqlBaseParser.InlineCastContext ctx) {
         Source source = source(ctx);
-        DataType dataType = typedParsing(this, ctx.dataType(), DataType.class);
+        // NOCOMMIT this seems like trouble
+        DataTypes dataType = typedParsing(this, ctx.dataType(), DataTypes.class);
```

Comment:
It's deeply unclear to me why we're passing in `DataType.class` here; It reads like we're checking that the type parsing returns a type? but that doesn't make sense, so I assume these undocumented methods are doing something non-obvious.